firebase-common/src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -110,6 +111,9 @@
   private final FirebaseOptions options;
   private final ComponentRuntime componentRuntime;
 
+  private final AtomicBoolean emulatorSettingsFrozen = new AtomicBoolean(false);
+  private EmulatorSettings emulatorSettings = EmulatorSettings.getDefault();
```

Comment:
nit: now that settings are immutable, consider sharing the default instance across app instances.

Reply:
Ah yeah good call!